dll/win32/imm32/ime.c

```diff
@@ -604,45 +613,38 @@ BOOL WINAPI ImmNotifyIME(HIMC hIMC, DWORD dwAction, DWORD dwIndex, DWORD_PTR dwV
 BOOL WINAPI
 ImmGetImeInfoEx(PIMEINFOEX pImeInfoEx, IMEINFOEXCLASS SearchType, PVOID pvSearchKey)
 {
-    HKL hKL;
-    if (SearchType == ImeInfoExKeyboardLayout || SearchType == ImeInfoExKeyboardLayoutTFS)
+    BOOL bTextServiceDisabled = FALSE;
+
+    if (SearchType == ImeInfoExKeyboardLayoutTFS)
     {
-        hKL = *(HKL*)pvSearchKey;
-        pImeInfoEx->hkl = hKL;
+        SearchType = ImeInfoExKeyboardLayout;
+        bTextServiceDisabled = CtfImmIsTextFrameServiceDisabled();
+    }
 
-        if (SearchType == ImeInfoExKeyboardLayoutTFS)
-        {
-            if (!IS_IME_HKL(hKL))
-            {
-                if (CtfImmIsTextFrameServiceDisabled() || !IS_CICERO_MODE() || IS_16BIT_MODE())
-                {
-                    TRACE("\n");
-                    return FALSE;
-                }
-            }
+    if (SearchType == ImeInfoExKeyboardLayout)
+    {
+        HKL hKL = *(HKL *)pvSearchKey;
+        pImeInfoEx->hkl = hKL;
```

Comment:
Should this be done unconditionally, even if this caller-given hKL is _not_ an IME hKL?
If not, the `IS_IME_HKL(hKL)` test should probably be done before.

Reply:
> Should this be done unconditionally, even if this caller-given hKL is not an IME hKL?

Yes, it should.